rholang/src/test/scala/coop/rchain/rholang/interpreter/RegistrySpec.scala

```diff
@@ -824,33 +541,138 @@ class RegistrySpec extends FlatSpec with Matchers with RegistryTester {
 
     val expectedUri = Registry.buildURI(uriBytes)
 
-    result.get(resultChanList("result0")) should be(
-      Some(
-        Row(
-          List(
-            Datum.create[Par, ListParWithRandom](
-              GString("result0"),
-              ListParWithRandom(Seq(GUri(expectedUri)), randResult0),
-              false
-            )
-          ),
-          List()
-        )
-      )
-    )
-    result.get(resultChanList("result1")) should be(
-      Some(
-        Row(
-          List(
-            Datum.create[Par, ListParWithRandom](
-              GString("result1"),
-              ListParWithRandom(Seq(expectedBundle), randResult1),
-              false
-            )
-          ),
-          List()
-        )
-      )
+    checkResult(result, "result0", GUri(expectedUri), randResult0)
+    checkResult(result, "result1", expectedBundle, randResult1)
+  }
+
+  "Signed Insert" should "work like plain insert if the signatures match" in {
+    // Secret key:
+    // d039d5c634ad95d968fc18368d81b97aaecd32fc7cf6eec07a97c5ac9f9fcb5b11afb9a5fa2b3e194b701987b3531a93dbdf790dac26f8a2502cfa5d529f6b4d
+    val registerString =
+      """
+      new rr(`rho:registry:insertSigned:ed25519`), rl(`rho:registry:lookup`), ack in {
+        rr!("11afb9a5fa2b3e194b701987b3531a93dbdf790dac26f8a2502cfa5d529f6b4d".hexToBytes(),
+            (789, "entry"),
+            "20c3b7da06565933400cb61301ffa14df82ef09b046c8152e02e8047d6f69ee2c2a2e4114db7ceb01eb828dfc98c15e40a502f9d85c58ca03734cab549e85e0d".hexToBytes(),
+            *ack) |
```

Comment:
Pubkey,
Tuple of (nonce, value)
Signature,
Return channel.

Reply:
Can you add the `pubKey`, `secKey` to the tests. Also the hashes and signatures so it is explicit in the test.

Reply:
The secret key is already there in the comments
I added the public key.
There are no hashes.
The signatures are over the serialized form of the terms in question.

Reply:
I would much rather have the signature in the code.

Reply:
@pyrocto I strongly disagree here. I don't want to interpolate a string that we then test. I also disagree that it's non-obvious. The public key is already in the comments, And the other string must be the signature. I think that makes point 1 above clear.

As to point 2. I specifically didn't get these values by using scala.
I serialized the representation to hex at the sbt console, and then signed them using a completely different ed25519 implementation. Just like how in the crypto tests, I used a patched version of the blake2 reference code to come up with the hashes.

Reproducing my steps may or may not be desirable given the above.

Reply:
With all due respect, @pyrocto is not the best person to resolve this dispute in my opinion, because both of you have knowledge about crypto, and how the registry should work, above average. Another fact that support this claim is that both of you already knew how to interact with the registry even though it was not yet documented. That makes you biased as to whether things are non-obvious or not.

Reply:
We need a description somewhere of how to generate the numbers so that if something changes and we have to update the test, we don't have to reverse engineer anything.  The process should be documented in the test.  If you do it in the test and interpolate, the documentation is part of the test.  I don't think that's always desirable, since checking against a different implementation adds to the confidence.  If you patched the blake2 reference code, that should be checked in somewhere so others can use it.  If there's a way to do it at sbt console, that should be described in the test.

Reply:
@goral09 tag Whomever you want.

Reply:
@goral09 Does my comment at least help answer the questions you had?

Reply:
It is better. Let's wait until noon your time for comments and if no one will answer I will approve your PR.